Orbit Menu: Add Auto-Observe mode! Refresh button! Alive/Ghost Counters! No more [ghost] tag!

MrDoomBringer opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
adds an auto-observe mode, something I've been wanting to do for a long time! Makes it so anything you orbit you observe as well. A button in the orbit menu toggles this on or off.

Also, adds a refresh button to the orbit menu. Not only is this performant - because we have a recursive orbitcounter now - but it also prevents you from accidently clicking on new mobs as the menu updates.

Also does some minor code improvement, ie. changing a couple 1s and 0s to TRUEs and FALSEs.
Finally, this fixes some bugs that mothblocks pointed out to me, shoutout to mothblocks

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
orbit menu....better....
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: The orbit menu now has an Auto-Observe button! No more sifting through the lame observe menu to snoop in people's backpacks! Also, you can refresh the orbit menu now.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Jared-Fogle commented on tgui/packages/tgui/interfaces/Orbit.js:
```diff
@@ -1,10 +1,9 @@
 import { resolveAsset } from '../assets';
 import { useBackend, useLocalState } from '../backend';
-import { Box, Button, Flex, Icon, Input, Section } from '../components';
+import { Box, Button, Flex, Icon, Input, Section, Divider } from '../components';
```
```suggestion
import { Box, Button, Divider, Flex, Icon, Input, Section } from '../components';
```

Jared-Fogle commented on tgui/packages/tgui/interfaces/Orbit.js:
```diff
@@ -86,6 +85,7 @@ export const Orbit = (props, context) => {
     ghosts,
     misc,
     npcs,
+    auto_observe,
```
Alphabetical order

Jared-Fogle commented on tgui/packages/tgui/interfaces/Orbit.js:
```diff
@@ -161,7 +181,7 @@ export const Orbit = (props, context) => {
         )}
 
-        <Section title="Alive">
+        <Section title={"Alive - (" + alive.length + ")"}>
```
```suggestion
        <Section title={`Alive - (${alive.length})`}>
```

Jared-Fogle commented on tgui/packages/tgui/interfaces/Orbit.js:
```diff
@@ -173,7 +193,7 @@ export const Orbit = (props, context) => {
         </Section>
 
-        <Section title="Ghosts">
+        <Section title={"Ghosts - (" + ghosts.length + ")"}>
```
```suggestion
        <Section title={`Ghosts - (${ghosts.length})`}>
```

Jared-Fogle commented on tgui/packages/tgui/interfaces/Orbit.js:
```diff
@@ -140,6 +140,26 @@ export const Orbit = (props, context) => {
+                tooltipPosition="bottom-left"
+                selected={auto_observe}
+                icon={auto_observe === 1 ? "toggle-on" : "toggle-off"}
+                onClick={() => act('toggle_observe')} />
```
It doesn't matter, but I just want the file to stay consistent.

```suggestion
                onClick={() => act("toggle_observe")} />
```

Jared-Fogle commented on tgui/packages/tgui/interfaces/Orbit.js:
```diff
@@ -140,6 +140,26 @@ export const Orbit = (props, context) => {
+                tooltip="Refresh"
+                tooltipPosition="bottom-left"
+                icon="sync-alt"
+                onClick={() => act('refresh')} />
```
```suggestion
                onClick={() => act("refresh")} />
```

Jared-Fogle commented on tgui/packages/tgui/interfaces/Orbit.js:
```diff
@@ -140,6 +140,26 @@ export const Orbit = (props, context) => {
+                tooltip="Toggle Auto-Observe"
+                tooltipPosition="bottom-left"
+                selected={auto_observe}
+                icon={auto_observe === 1 ? "toggle-on" : "toggle-off"}
```
```suggestion
                icon={auto_observe ? "toggle-on" : "toggle-off"}
```

I like to pretend that BYOND has real booleans.

ATH1909 commented:
By the way, how DO you look in someone's backpack while you're observing them? I haven't been able to figure out how to do that yet...

Jared-Fogle commented:
@ATH1909 Alt click.

MrDoomBringer commented:
@Jared-Fogle Clicking normally works as well

Timberpoes reviewed (changes requested):
```
Error: our build differs from the build committed into git.

Please rebuild tgui.
```

Please rebase and rebuild tgui before this can be merged.

MrDoomBringer commented:
rebased and rebuilded tgui before this is merged

stylemistake commented:
Rebuild and ping me to merge.

MrDoomBringer committed:
Observe, Refresh, alive/ghost count, and no more [ghost]


tgui moment


Apply suggestions from mr jaredfogle

Co-authored-by: Jared-Fogle <35135081+Jared-Fogle@users.noreply.github.com>
mister jaredfogle


yes? he built?
tgui

MrDoomBringer committed:
tgui
